exist-core/src/main/java/org/exist/xquery/functions/array/ArrayType.java

```diff
@@ -382,6 +382,30 @@ public static Sequence flatten(Sequence input) throws XPathException {
         return flatten ? flatten(input, new ValueSequence(input.getItemCount() * 2)) : input;
     }
 
+    @Override
+    public boolean containsReference(final Item item) {
+        for (int i = 0; i < vector.length(); i++) {
+            final Sequence value = vector.nth(i);
+            if (value == item) {
+                return true;
+            }
+            return value.containsReference(item);
+        }
+        return false;
+    }
+
+    @Override
+    public boolean contains(final Item item) {
+        for (int i = 0; i < vector.length(); i++) {
+            final Sequence value = vector.nth(i);
+            if (value.equals(item)) {
+                return true;
+            }
+            return value.contains(item);
+        }
+        return false;
```

Comment:
Same here:
```
if (value.equals(item) || value.containsReference(item)) {
  return true;
}
```

Reply:
AFAIK the logic you have suggested is exactly the same as what is already there. What am I missing?

Reply:
Okay fixed.